Predictor_submission_template.ipynb

```diff
@@ -369,15 +369,160 @@
-    "        print(error)\n",
-    "else:\n",
-    "    print(\"All good!\")"
+    "def validate(start_date, end_date, ip_file, output_file):\n",
+    "    # First, delete any potential old file\n",
+    "    try:\n",
+    "        os.remove(output_file)\n",
+    "    except OSError:\n",
+    "        pass\n",
+    "    \n",
+    "    # Then generate the prediction, calling the official API\n",
+    "    !python predict.py -s {start_date} -e {end_date} -ip {ip_file} -o {output_file}\n",
+    "    \n",
+    "    # And validate it\n",
+    "    errors = validate_submission(start_date, end_date, ip_file, output_file)\n",
+    "    if errors:\n",
+    "        for error in errors:\n",
+    "            print(error)\n",
+    "    else:\n",
+    "        print(\"All good!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4 days, no gap\n",
+    "- All countries and regions\n",
+    "- Official number of cases is known up to start_date\n",
+    "- Intervention Plans are the official ones"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "validate(start_date=\"2020-08-01\",\n",
+    "         end_date=\"2020-08-04\",\n",
+    "         ip_file=\"../../../validation/data/2020-09-30_historical_ip.csv\",\n",
+    "         output_file=\"predictions/val_4_days.csv\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1 month in the future\n",
+    "- 2 countries only\n",
+    "- there's a gap between date of last known number of cases and start_date\n",
+    "- For future dates, Intervention Plans contains scenarios for which predictions are requested to answer the question: what will happen if we apply these plans?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "validate(start_date=\"2021-01-01\",\n",
+    "         end_date=\"2021-01-31\",\n",
+    "         ip_file=\"../../../validation/data/future_ip.csv\",\n",
+    "         output_file=\"predictions/val_1_month_future.csv\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 180 days, from a future date, all countries and regions\n",
+    "- Prediction start date is 1 week from now. (i.e. assuming submission date is 1 week from now)  \n",
+    "- Prediction end date is 6 months after start date.  \n",
+    "- Prediction is requested for all available countries and regions.  \n",
+    "- Intervention plan scenario: freeze last known intervention plans for each country and region.  \n",
+    "\n",
+    "As the number of cases is not known yet between today and start date, but the model relies on them, the model has to predict them in order to use them.  \n",
+    "This test is the most demanding test. It should take less than 1 hour to generate the prediction file."
```

Comment:
Could we have a timer and fail if > 1 hour? Or issue a warning?

Reply:
Not easily...